src/AlgebraicGeometry/TropicalGeometry/groebner_fan.jl

```diff
@@ -0,0 +1,544 @@
+    # try finding any positive vector inside of it
+    positiveOrthant = positive_hull(identity_matrix(ZZ,n))
+    posHomogeneitySpace = intersect(homogeneitySpace,positiveOrthant)
+    homogeneityVector = unique_identifying_point(posHomogeneitySpace)
+    if isnothing(findfirst(vi -> (vi<=0), homogeneityVector))
+        return homogeneityVector
+    end
+
+    # return nothing if everything fails
+    return nothing
+end
+
+
+
+@doc raw"""
+    maximal_groebner_cone(G::Vector{<:MPolyElem}, ord::MonomialOrdering, homogeneityWeight::Union{Nothing,Vector{fmpz}}=nothing)
+
+# Input
+- `G`: A reduced Groebner basis of an ideal `I` with respect to `ord`,
+- `ord`: A global monomial ordering,
+- `homogeneityWeight`: Either `Nothing` or a positive weight vector under which `I` is weighted homogeneous.
+
+# Output
+A `Cone` that is the maximal Groebner cone of `I` with respect to monomial ordering `ord`.  If `homogeneityWeight==nothing`, the Groebner cone will be contained in the positive orthant.  If `homogeneityWeight!=nothing`, the Groebner cone will not be restricted to the positive orthant and have a lineality space that contains `homogeneityWeight`.
+
+# Examples
+```jldoctest
+julia> Qx,(x1,x2,x3) = PolynomialRing(QQ,["x1","x2","x3"])
+(Multivariate Polynomial Ring in x1, x2, x3 over Rational Field, QQMPolyRingElem[x1, x2, x3])
+
+julia> Qx,(x1,x2,x3) = PolynomialRing(QQ,["x1","x2","x3"])
+(Multivariate Polynomial Ring in x1, x2, x3 over Rational Field, QQMPolyRingElem[x1, x2, x3])
+
+julia> G = [x1,x2+x3]
+2-element Vector{QQMPolyRingElem}:
+ x1
+ x2 + x3
+
+julia> ord = lex(Qx)
+lex([x1, x2, x3])
+
+julia> rays_modulo_lineality(maximal_groebner_cone(G,ord))
+(rays_modulo_lineality = RayVector{QQFieldElem}[[1, 0, 0], [0, 1, 1], [0, 1, 0]], lineality_basis = RayVector{QQFieldElem}[])
+
+julia> w = homogeneity_vector(G)
+3-element Vector{ZZRingElem}:
+ 1
+ 1
+ 1
+
+julia> rays_modulo_lineality(maximal_groebner_cone(G,ord,w))
+(rays_modulo_lineality = RayVector{QQFieldElem}[[0, 0, -1]], lineality_basis = RayVector{QQFieldElem}[[1, 0, 0], [0, 1, 1]])
+
+```
+"""
+function maximal_groebner_cone(G::Vector{<:MPolyElem}, ord::MonomialOrdering)
+    # calling with no homogeneityWeight
+    # is calling with `nothing` as homogeneityWeight
+    return maximal_groebner_cone(G,ord,nothing)
+end
+function maximal_groebner_cone(G::Vector{<:MPolyElem}, ord::MonomialOrdering, homogeneityWeight::Nothing)
+    # calling with `nothing` as homogeneityWeight
+    # means that C needs to be restricted to the positive orthant
+    C = maximal_groebner_cone_extended(G,ord)
+    positiveOrthant = positive_hull(identity_matrix(ZZ,ambient_dim(C)))
+    return intersect(C,positiveOrthant)
+end
+function maximal_groebner_cone(G::Vector{<:MPolyElem}, ord::MonomialOrdering, homogeneityWeight::Vector{fmpz})
+    # calling with a positive homogeneityWeight
+    # means that C does not be restricted to the positive orthant
+    return maximal_groebner_cone_extended(G,ord)
+end
```

Comment:
Here I am again have to ask: Instead of `nothing` wouldn't it be easier to have an all zeroes vector?